### DISCUSSION TITLE
Which version of eclipse.jdt.ls is contained in vscode-java

### DISCUSSION THREAD

**Discussion by sailingKieler** [redhat-developer/vscode-java#3066] [Q&A] [answered]
Hi,

which version of https://github.com/eclipse/eclipse.jdt.ls contained in this extension? Current version is v1.16.0.
Since the tag names are not in sync: Is there a source specifying for each release the jdt.ls (tag) version?

best regards,
Christian

**Comment by fbricon** [comment, marked as answer]
There's currently no published mapping of vscode-java <-> jdt.ls versions. vscode-java pakaging rebuilds jdt.ls from source (master) during releases.
The best way to check which jdt.ls commit went into vscode-java is to :
- extract the redhat.java.vsix (or look under ~/.vscode/extensions/redhat.java-version)
- unzip server/plugins/org.eclipse.jdt.ls.core_version.jar
- open META-INF/MANIFEST.MF, look for the `Eclipse-SourceReferences` entry. You should see something like:

> Eclipse-SourceReferences: scm:git:https://github.com/eclipse/eclipse.jdt
 .ls.git;path="org.eclipse.jdt.ls.core";commitId=6d94ed2ff88d9678e74d121
 bf37719cf5e9e8ef9

**Reply by sailingKieler** [reply]
I didn't expect that, thank's a lot for this insight @fbricon. 🙏
